gateway/src/apicast/metrics/3scale_backend_calls.lua

```diff
@@ -0,0 +1,36 @@
+local prometheus = require('apicast.prometheus')
+
+local format = string.format
+
+local _M = {}
+
+local threescale_backend_call = prometheus(
+  'counter',
+  'threescale_backend_calls',
+  "Calls to the 3scale backend",
+  { 'endpoint', 'status' }
+)
+
+local function label_for_status(status)
+  if not status or status == '' or status == 0 then
+    return 'invalid_status'
+  else
+    return format("%dxx", status/100)
+  end
+end
+
+local valid_endpoint = {
+  auth = true, authrep = true, report = true
+}
+
+function _M.report(endpoint, status)
+  if threescale_backend_call then
+    if valid_endpoint[endpoint] then
+      threescale_backend_call:inc(1, { endpoint, label_for_status(status) })
+    else
+      ngx.log(ngx.WARN, 'Invalid 3scale backend endpoint endpoint when updating metrics')
```

Comment:
Do we really care about this?

Reply:
What about the possibility of having invalid information in Prometheus?

Reply:
But what would be that invalid information? Someone part of code calling the wrong endpoint on the backend. I don't consider that invalid metric. If that call happened, then it should be measured and reported.

Reply:
Fair enough.